### PR TITLE
test: Add more tests to active executions file

### PR DESCRIPTION
**Pull request by krynble**

### CODE REVIEW

File: packages/cli/test/unit/ActiveExecutions.test.ts [hunk 1]
```diff
@@ -66,6 +66,55 @@ describe('ActiveExecutions', () => {
...
+		const newExecution = mockExecutionData();
+		const executionId = await activeExecutions.add(newExecution);
+		const postExecutePromise = activeExecutions.getPostExecutePromise(executionId);
+		// Force the above to be executed since we cannot await it
+		await new Promise((res) => {
+			setTimeout(res, 100);
+		});
+		const fakeOutput = mockFullRunData();
+		activeExecutions.remove(executionId, fakeOutput);
+		expect(postExecutePromise).resolves.toEqual(fakeOutput);
```
Comment (ivov) [inline packages/cli/test/unit/ActiveExecutions.test.ts:96]: Nice, I didn't know about `.resolves`

File: packages/cli/test/unit/ActiveExecutions.test.ts [hunk 2]
```diff
@@ -66,6 +66,55 @@ describe('ActiveExecutions', () => {
...
+	});
+
+	// This test contains a mix of promise + async error throw that I was unable to figure out
+	// how to test.
+	xtest('Should throw error when trying to create a promise with invalid execution', async() => {
+		expect(() => {
+				activeExecutions.getPostExecutePromise(FAKE_EXECUTION_ID);
+		}).toThrow();
```
Comment (ivov) [inline packages/cli/test/unit/ActiveExecutions.test.ts:104]: According to the docs this should be it?
```suggestion
await expect(activeExecutions.getPostExecutePromise(FAKE_EXECUTION_ID)).rejects.toThrow(
	'My error message',
);
```

File: packages/cli/test/unit/ActiveExecutions.test.ts [hunk 3]
```diff
@@ -83,9 +132,25 @@ function mockExecutionData(): IWorkflowExecutionDataProcess {
...
+
 function mockCancelablePromise(): PCancelable<IRun> {
 	return new PCancelable(async (resolve) => {
 		resolve();
 	});
 }
 
+async function mockDeferredPromise(): Promise<IDeferredPromise<IExecuteResponsePromiseData>> {
+	return await createDeferredPromise<IExecuteResponsePromiseData>();
+}
```
Comment (ivov) [inline packages/cli/test/unit/ActiveExecutions.test.ts:155]:
```suggestion
function mockDeferredPromise(): Promise<IDeferredPromise<IExecuteResponsePromiseData>> {
	return createDeferredPromise<IExecuteResponsePromiseData>();
}
```

**Review by ivov — COMMENTED** [review]
One pattern that I like is separating the arrange-act-assert steps with blank lines, as it makes the structure of each test more readable/glanceable.

File: packages/cli/test/unit/ActiveExecutions.test.ts (hunk 3 above)
Comment (krynble) [reply packages/cli/test/unit/ActiveExecutions.test.ts:155]: Thanks, applied!

File: packages/cli/test/unit/ActiveExecutions.test.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { mocked } from 'jest-mock';
 import PCancelable from 'p-cancelable';
 import { v4 as uuid } from 'uuid';
 import { createDeferredPromise, IDeferredPromise, IExecuteResponsePromiseData, IRun } from 'n8n-workflow';
+import { rejects } from 'assert';
```
Comment (ivov) [inline packages/cli/test/unit/ActiveExecutions.test.ts:6]: Is this import needed? Looks autocompleted by accident.

Comment (krynble) [reply packages/cli/test/unit/ActiveExecutions.test.ts]: Thanks for spotting!

**Review by ivov — APPROVED** [review]
🚀

**Pull request merged**